sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -56,6 +56,11 @@ class InMemoryFileIndex(
   extends PartitioningAwareFileIndex(
     sparkSession, parameters, userSpecifiedSchema, fileStatusCache) {
 
+  assert(userSpecifiedPartitionSpec.isEmpty ||
+    userSpecifiedPartitionSpec.get.partitions.map(_.path).equals(rootPathsSpecified),
+    s"The rootPathsSpecified ($rootPathsSpecified) is inconsistent with the file paths " +
+      s"of userSpecifiedPartitionSpec (${userSpecifiedPartitionSpec.get.partitions.map(_.path)}).")
+
```

Comment:
Just add an assertion incidentally here, this is not related to this PR.
if not ok, i can remove it.